webapp/graphite/render/hashing.py

```diff
@@ -43,25 +61,32 @@ def compactHash(string):
 
 
 class ConsistentHashRing:
-  def __init__(self, nodes, replica_count=100):
+  def __init__(self, nodes, replica_count=100, hash_type='carbon_ch'):
     self.ring = []
     self.ring_len = len(self.ring)
     self.nodes = set()
     self.nodes_len = len(self.nodes)
     self.replica_count = replica_count
+    self.hash_type = hash_type
     for node in nodes:
       self.add_node(node)
 
   def compute_ring_position(self, key):
-    big_hash = md5( str(key) ).hexdigest()
-    small_hash = int(big_hash[:4], 16)
+    if self.hash_type == 'fnv1a_ch':
+      big_hash = '{:x}'.format(int(fnv32a( str(key) )))
+      small_hash = int(big_hash[:4], 16) ^ int(big_hash[4:], 16)
+    else:
+      big_hash = md5(str(key)).hexdigest()
+      small_hash = int(big_hash[:4], 16)
     return small_hash
 
   def add_node(self, key):
     self.nodes.add(key)
     self.nodes_len = len(self.nodes)
     for i in range(self.replica_count):
       replica_key = "%s:%d" % (key, i)
+      if self.hash_type == 'fnv1a_ch':
+        replica_key = "%d-%s" % (i, key[1])
```

Comment:
I would prefer an explicit `if else` here unless this is considered more pythonic.